lib/zendesk_apps_support/validations/secrets.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+
+module ZendeskAppsSupport
+  module Validations
+    module Secrets
+      SECRET_KEYWORDS = %w[
+        pass password secret secretToken secret_token auth_key
+        authKey auth_pass authPass auth_user AuthUser username api_key
+      ].freeze
+
+      class << self
+        def call(package)
+          compromised_files = []
+
+          # Rely on the regexes used by truffleHog (https://github.com/dxa4481/truffleHog) to detect application secrets
+          trufflehog_regexes = JSON.parse(Net::HTTP.get(URI.parse('https://raw.githubusercontent.com/dxa4481/truffleHogRegexes/master/truffleHogRegexes/regexes.json'))).freeze
```

Comment:
Seems silly to make a network request for this at runtime, though I wasn't sure of the correct way to include these when compiling the gem. Open to suggestions here.

Reply:
We could patch `bump/tasks` to download a new version of the file when a new version of the gem is bumped. I'm in favour of hardcoding the file anyway so that if that download fails we at least have something.